core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/ConsumerBindingService.java

```diff
@@ -64,9 +66,8 @@
     return save(consumerBinding);
   }
 
-  @PreAuthorize("hasPermission(#consumerBindingKey, 'UPDATE_STATUS')")
-  public Optional<ConsumerBinding> updateStatus(ConsumerBindingKey consumerBindingKey, Status status) {
-    ConsumerBinding consumerBinding = read(consumerBindingKey).get();
+  @PreAuthorize("hasPermission(#consumerBinding, 'UPDATE_STATUS')")
+  public Optional<ConsumerBinding> updateStatus(ConsumerBinding consumerBinding, Status status) {
```

Comment:
Updating the binding status shouldn't need the whole binding?

Reply:
The RangerAuthoriser contract needs the full object (Entity<*>).
This logic has been pushed up to the graphql layer?